tests/server.spec.js

```diff
@@ -336,6 +336,35 @@ describe("API", () => {
       }
     });
 
+    describe("Changing the fabric process", () => {
+      const jeanQuery = [
+        "mass=0.45",
+        "product=jean",
+        "fabricProcess=weaving",
+        "materials[]=ei-coton;1",
+        "countryFabric=TR",
+        "countryDyeing=TR",
+        "countryMaking=TR",
+        "fading=true",
+      ];
+
+      it("should change the waste", async () => {
+        let response = await makeRequest("/api/textile/simulator/detailed", jeanQuery);
+        expectStatus(response, 200);
+        fabricLifeCycle = response.body.lifeCycle.find((l) => l.label == "Tissage & Tricotage");
+        expect(fabricLifeCycle.waste).toEqual(0.045271493212669676);
+
+        const jeanQueryKnittingMix = jeanQuery.map((input) =>
+          input == "fabricProcess=weaving" ? "fabricProcess=knitting-mix" : input,
+        );
+
+        response = await makeRequest("/api/textile/simulator/detailed", jeanQueryKnittingMix);
+        expectStatus(response, 200);
+        fabricLifeCycle = response.body.lifeCycle.find((l) => l.label == "Tissage & Tricotage");
+        expect(fabricLifeCycle.waste).toEqual(0.039095022624434386);
```

Comment:
Maybe what we really want is simply checking that the waste is *less* than with the previous process used? So we won't have to maintain the exact numbers here